zkevm-circuits/src/evm_circuit/execution/error_return_data_oo_bound.rs

```diff
@@ -84,13 +83,23 @@ impl<F: Field> ExecutionGadget<F> for ErrorReturnDataOutOfBoundGadget<F> {
             return_data_length.expr(),
             from_bytes::expr(&end.cells[..N_BYTES_U64]),
         );
-        // Any of [offset_out_of_range, end_out_of_range, end_exceed_length] occurs.
-        cb.require_in_set(
-            "Any of [offset_out_of_range, end_out_of_range, end_exceed_length] occurs",
-            not::expr(is_data_offset_within_range.expr())
-                + not::expr(is_end_within_range.expr())
-                + is_end_exceed_length.expr(),
-            vec![1.expr(), 2.expr(), 3.expr()],
+
+        // `end > U256::MAX` is necessary to check with AddWordsGadget carry. Since
+        // offset plus length may exceed a Word and remainder (end) may be less
+        // than `u64::MAX` (e.g. Word::MAX + 1).
+        cb.require_equal(
+            "Any of [offset > u64::MAX, end > U256::MAX, remainder_end > u64::MAX, remainder_end > return_data_length] occurs",
+            or::expr([
+                // offset > u64::MAX
+                not::expr(is_data_offset_within_range.expr()),
+                // end > U256::MAX
```

Comment:
`data + offset > U256::MAX` seems more self-explained

Reply:
Fixed to `data_offset + size > U256::MAX` in [4ea268b](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1321/commits/4ea268b414ea757e452eab267e9a8a656c0cadcc).